File: Python/Dosen.py
from Sivitas import Sivitas

class Dosen(Sivitas): 
    __nip = ""
    __pendTerakhir = ""
    __keahlian = ""

    def __init__(self):
        super(Dosen, self).__init__()
        self.nip = ""
        self.pendTerakhir =""
        self.keahlian = ""

    def setNip(self, a):
        self.nip = a
    def setPendTerakhir(self, a):
        self.pendTerakhir = a
    def setKeahlian(self, a):
        self.keahlian = a
    
    def getNip(self):
        return self.nip
    def getPendTerakhir(self):
        return self.pendTerakhir
    def getKeahlian(self):
        return self.keahlian
    
    

File: CPP/Dosen.cpp
#include <iostream>
#include <string>
#include "Sivitas.cpp"

using namespace std;

class Dosen : public Sivitas
{
private:
    string nip;
    string pendTerakhir;
    string keahlian;

public:
    Dosen()
    {
        setNik("");
        setNama("");
        setJk("");
        setAsalUniv("");
        setEmailEdu("");
        nip = "";
        pendTerakhir = "";
        keahlian = "";
    };
    Dosen(string a, string b, string c, string d, string e, string f, string g, ProgramStudi h, string i)
    {
        setNik(a);
        setNama(b);
        setJk(c);
        setAsalUniv(d);
        setEmailEdu(e);
        nip = f;
        pendTerakhir = g;
        keahlian = i;
        setProdi(h);
    }

    void setNip(string a)
    {
        nip = a;
    }
    void setPendTerakhir(string a)
    {
        pendTerakhir = a;
    }
    void setKeahlian(string a)
    {
        keahlian = a;
    }

    string getnip()
    {
        return nip;
    }
    string getPendTerakhir()
    {
        return pendTerakhir;
    }
    string getKeahlian()
    {
        return keahlian;
    }

    ~Dosen()
    {
    }
};
